pkg/ca/fileca/load.go

```diff
@@ -27,21 +27,64 @@ import (
 	"go.step.sm/crypto/pemutil"
 )
 
-func loadKeyPair(certPath, keyPath, keyPass string) (*x509.Certificate, crypto.Signer, error) {
+func loadKeyPair(certPath, keyPath, keyPass string) ([]*x509.Certificate, crypto.Signer, error) {
 
 	var (
-		cert *x509.Certificate
-		err  error
-		key  crypto.Signer
+		certs []*x509.Certificate
+		err   error
+		key   crypto.Signer
 	)
 
-	// TODO: Load chain of certs (intermediates and root) instead of just one
-	// certificate.
-	cert, err = pemutil.ReadCertificate(certPath)
+	// NB: certs are ordered from leaf at certs[0] to root at
+	// certs[len(certs)-1]
+	certs, err = pemutil.ReadCertificateBundle(certPath)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	// Verify certificate chain
+	{
+		roots := x509.NewCertPool()
+		roots.AddCert(certs[len(certs)-1])
+
+		intermediates := x509.NewCertPool()
+		if len(certs) > 2 {
```

Comment:
Should this be `if len(certs) > 1`?

Reply:
I don't think so. In the case of `len(certs) == 2` we add `certs[1]` to the root pool and then verify `certs[0]` and the pool of intermediates is empty. Unless perhaps I misunderstood your previous comments. You don't want to add `certs[0]` to the pool of intermediates of that is the certificate your verifying right?

That said, I don't actually have a test cast for `len(certs) == 2` so I'll add that just to be certain.

Reply:
Never mind, this is good! Forgot this was verifying the intermediate. Yea, you don't need to add certs[0] to the pool.

Reply:
Added test data and test case for `len(certs) == 2` 🚀